docs/userguide/CDN/USERGUIDE.md

```diff
@@ -0,0 +1,289 @@
+
+* **Status**: The status indicates the current state of the service. The time it takes for a service configuration to be distributed amongst a CDN provider cache can vary.
+
+* **Purge**: Purging removes content from the edge servers - thus invalidating the content - so that it can be refreshed from your origin servers.
+        
+* **Caching rule**: A caching rule provides you with fine-grained control over the time-to-live (TTL) of an object. When the TTL expires for an object, the edge node pulls the object from the origin again.
+
+* **Restriction**: A restriction enables you to define rules about who can or cannot access content from the cache. Examples of a restriction are allowing requests only from certain domains, geographies, or IP addresses.
+
+
+## Prerequisites
+
+### Client
+To use the CDN service, you must first instantiate a `OpenStack` or `Rackspace` client object.
+
+* If you are working with an OpenStack cloud, instantiate an `OpenCloud\OpenStack` client as follows:
+
+    ```php
+    use OpenCloud\OpenStack;
+
+    $client = new OpenStack('<OPENSTACK CLOUD IDENTITY ENDPOINT URL>', array(
+        'username' => '<YOUR OPENSTACK CLOUD ACCOUNT USERNAME>',
+        'password' => '<YOUR OPENSTACK CLOUD ACCOUNT PASSWORD>'
+    ));
+    ```
+
+* If you are working with the Rackspace cloud, instantiate a `OpenCloud\Rackspace` client as follows:
+
+    ```php
+    use OpenCloud\Rackspace;
+
+    $client = new Rackspace(Rackspace::US_IDENTITY_ENDPOINT, array(
+        'username' => '<YOUR RACKSPACE CLOUD ACCOUNT USERNAME>',
+        'apiKey'   => '<YOUR RACKSPACE CLOUD ACCOUNT API KEY>'
+     ));
+    ```
+
+### CDN service
+
+All CDN operations are done via an _CDN service object_. To
+instantiate this object, call the `cdnService` method on the `$client`
+object. This method takes one argument:
+
+| Position | Description | Data type | Required? | Default value | Example value |
+| -------- | ----------- | ----------| --------- | ------------- | ------------- |
+|  1       | Name of the service, as it appears in the service catalog | String | No | `null`; automatically determined when possible | `rackCDN` |
+
+
+```php
+$cdnService = $client->cdnService();
+```
+
+## Services
+
+A service represents your web application that has its content cached to the edge nodes.
+
+### Create a service (to represent your web application)
+
+
+This operation takes one parameter, an associative array, with the following keys:
+
+| Name | Description | Data type | Required? | Default value | Example value |
+| ---- | ----------- | --------- | --------- | ------------- | ------------- |
+| `name` | A human-readable name for the service. This name must be unique. | String | No | `null` | `acme_site` |
+| `domains` | An array of associative arrays, each specifying a domain name for your service. | Array of associative arrays | Yes | `null` | `array( array( 'domain' => 'www.acme.com' ), array ( 'domain' => 'acme.com' ) )` |
```

Comment:
Can wildcards be used for the name? Are there any other array keys supported apart from 'name'?